src/components/structures/GroupView.js

```diff
@@ -246,8 +252,36 @@ export default React.createClass({
         });
     },
 
+    _onAvatarSelected: function(e) {
+    },
+
+    _onAvatarChange: function(newAvatarUrl) {
+        const newProfileForm = Object.assign(this.state.profileForm, { avatar_url: newAvatarUrl });
+        this.setState({
+            profileForm: newProfileForm,
+        });
+    },
+
     _onSaveClick: function() {
-        // TODO: There's no API to edit group profile info yet.
+        this.setState({saving: true});
+        MatrixClientPeg.get().setGroupProfile(this.props.groupId, this.state.profileForm).then((result) => {
+            this.setState({
+                saving: false,
+                editing: false,
+                summary: false,
+            });
+            return this._loadGroupFromServer(this.props.groupId);
```

Comment:
`_loadGroupFromServer` doesn't appear to return a promise so the `return` is redundant here